sdk/python/feast/infra/aws.py

```diff
@@ -1,23 +1,165 @@
+import base64
 import os
 import uuid
 from datetime import datetime
 from pathlib import Path
 from tempfile import TemporaryFile
+from typing import Sequence, Union
 from urllib.parse import urlparse
 
 from colorama import Fore, Style
 
 import feast
-from feast.constants import AWS_LAMBDA_FEATURE_SERVER_IMAGE
+from feast import Entity, FeatureTable, FeatureView, __version__
+from feast.constants import (
+    AWS_LAMBDA_FEATURE_SERVER_IMAGE,
+    FEAST_USAGE,
+    SERVER_CONFIG_BASE64_ENV_NAME,
+)
 from feast.errors import S3RegistryBucketForbiddenAccess, S3RegistryBucketNotExist
 from feast.infra.passthrough_provider import PassthroughProvider
+from feast.infra.utils import aws_utils
 from feast.protos.feast.core.Registry_pb2 import Registry as RegistryProto
 from feast.registry_store import RegistryStore
 from feast.repo_config import RegistryConfig
 
+try:
+    import boto3
+except ImportError as e:
+    from feast.errors import FeastExtrasDependencyImportError
+
+    raise FeastExtrasDependencyImportError("aws", str(e))
+
 
 class AwsProvider(PassthroughProvider):
-    def _upload_docker_image(self) -> None:
+    def _get_lambda_name(self):
+        return f"feast-python-server-{__version__.replace('+', '_').replace('.', '_')}"
```

Comment:
Shouldn't this lambda name have the project name in it somewhere? I think that would be useful for debugging.